src/main/java/org/snomed/aag/data/services/CriteriaItemSignOffService.java

```diff
@@ -162,23 +180,59 @@ public void doCreateItems(Set<String> itemsToAccept, String branchPath, long hea
+		return true;
     }
 
-	public void deleteItems(Set<String> itemsToUnaccept, String branchPath, Integer projectIteration) {
-		logger.info("Deleting item sign-offs {} for branch {}, iteration {}", itemsToUnaccept, branchPath, projectIteration);
+	/**
+	 * Delete CriteriaItemSignOff for all given CriteriaItem.
+	 *
+	 * @param itemsToUnaccept           Identifiers of CriteriaItems to mark as incomplete.
+	 * @param branchPath                Branch path of CriteriaItems to mark as incomplete.
+	 * @param projectIteration          PK
+	 * @param projectAcceptanceCriteria Determine whether to mark Project Criteria or Task Criteria incomplete.
+	 */
+	public void deleteFrom(Set<String> itemsToUnaccept, String branchPath, Integer projectIteration, ProjectAcceptanceCriteria projectAcceptanceCriteria) {
+		logger.info("Deleting item sign-offs {} for branch {}, iteration {}", itemsToUnaccept, branchPath);
 		for (String itemId : itemsToUnaccept) {
-			repository.deleteByCriteriaItemIdAndBranchAndProjectIteration(itemId, branchPath, projectIteration);
+			doDeleteByCriteriaItemIdAndBranchAndProjectIteration(itemId, branchPath, projectIteration, projectAcceptanceCriteria);
+		}
+	}
+
+	// Find differently for PROJECT & TASK CriteriaItemSignOff
+	private List<CriteriaItemSignOff> doFindAllByBranchAndProjectIterationAndCriteriaItemIdIn(Collection<String> criteriaItemIdentifiers, String branchPath, Integer projectIteration, ProjectAcceptanceCriteria projectAcceptanceCriteria) {
+		boolean branchProjectLevel = projectAcceptanceCriteria.isBranchProjectLevel(branchPath);
+		if (branchProjectLevel) {
+			return repository.findAllByBranchAndProjectIterationAndCriteriaItemIdIn(branchPath, projectIteration, criteriaItemIdentifiers);
+		}
+
+		return repository.findAllByBranchAndCriteriaItemIdIn(branchPath, criteriaItemIdentifiers);
+	}
+
+	// Find differently for PROJECT & TASK CriteriaItemSignOff
+	private Optional<CriteriaItemSignOff> doFindByCriteriaItemIdAndBranchAndProjectIteration(String criteriaItemId, String branchPath, Integer projectIteration, ProjectAcceptanceCriteria projectAcceptanceCriteria) {
+		if (projectAcceptanceCriteria.isBranchProjectLevel(branchPath)) {
+			return repository.findByCriteriaItemIdAndBranchAndProjectIteration(criteriaItemId, branchPath, projectIteration);
+		}
+
+		return repository.findByCriteriaItemIdAndBranch(criteriaItemId, branchPath);
+	}
+
+	// Delete differently for PROJECT & TASK CriteriaItemSignOff
+	private void doDeleteByCriteriaItemIdAndBranchAndProjectIteration(String criteriaItemId, String branchPath, Integer projectIteration, ProjectAcceptanceCriteria projectAcceptanceCriteria) {
+		if (projectAcceptanceCriteria.isBranchProjectLevel(branchPath)) {
+			repository.deleteByCriteriaItemIdAndBranchAndProjectIteration(criteriaItemId, branchPath, projectIteration);
+		} else {
+			repository.deleteByCriteriaItemIdAndBranch(criteriaItemId, branchPath);
```

Comment:
These methods will be called from any context (when interacting with projects or tasks). I've added a condition to call the correct repository depending on whether the branch in question is the project or a task.

Reply:
Also, because these methods are so low level (within a repository service class) and I added an additional parameter, a lot of surrounding/calling code had to be changes which is basically why there are so many mild changes.

Reply:
These method names are very wordy for service level calls. I would expect wordy methods at the repository level because we have to talk about specific fields etc there, but at the service level it's good to name methods as their higher level functions. 
For example you could replace `doFindByCriteriaItemIdAndBranchAndProjectIteration` with `doFindTaskCriteriaItem`. The parameters of the method self document the fields that are required.

Reply:
Good point! I've made those changes and will keep that in mind for future.